lib/ts/controllers/s-modal.ts

```diff
@@ -0,0 +1,279 @@
+namespace Stacks {
+
+    export class ModalController extends Stacks.StacksController {
+        static targets = ["modal", "initialFocus"];
+
+        private modalTarget!: HTMLElement;
+        private initialFocusTarget!: HTMLElement;
+        private hasInitialFocusTarget!: boolean;
+
+        private _boundClickFn!: any;
+        private _boundKeypressFn!: any;
+
+        private returnElement!: HTMLElement;
+
+        private _boundTabTrap!: any;
+
+        connect () {
+            this.validate();
+        }
+
+        /**
+         * Disconnects all added event listeners on controller disconnect
+         */
+        disconnect() {
+            this.unbindDocumentEvents();
+        };
+
+        /**
+         * Toggles the visibility of the modal
+         */
+        toggle () {
+            this._toggle();
+        }
+
+        /**
+         * Shows the modal
+         */
+        show () {
+            this._toggle(true);
+        }
+
+        /**
+         * Hides the modal
+         */
+        hide () {
+            this._toggle(false);
+        }
+
+        /**
+         * Validates the modal settings and attempts to set necessary internal variables
+         */
+        private validate() {
+            // check for returnElement support
+            var returnElementSelector = this.data.get("return-element");
+            if (returnElementSelector) {
+                this.returnElement = <HTMLElement>document.querySelector(returnElementSelector);
+
+                if (!this.returnElement) {
+                    throw "Unable to find element by return-element selector: " + returnElementSelector;
+                }
+            }
+        }
+
+        /**
+         * Toggles the visibility of the modal element
+         * @param show Optional parameter that force shows/hides the element or toggles it if left undefined
+         */
+        private _toggle (show?: boolean | undefined) {
+            var toShow = show;
+            var isVisible = this.modalTarget.getAttribute("aria-hidden") === "false";
+
+            // if we're letting the class toggle, we need to figure out if the popover is visible manually
+            if (typeof toShow === "undefined") {
+                toShow = !isVisible;
+            }
+
+            // if the state matches the disired state, return without changing anything
+            if ((toShow && isVisible) || (!toShow && !isVisible)) {
+                return;
+            }
+
+            // show/hide events trigger before toggling the class
+            var triggeredEvent = this.triggerEvent(toShow ? "show" : "hide");
+
+            // if this pre-show/hide event was prevented, don't attempt to continue changing the modal state
+            if (triggeredEvent.defaultPrevented) {
+                return;
+            }
+
```

Comment:
```suggestion
            var triggeredEvent = this.triggerEvent(toShow ? "show" : "hide", { returnElement: this.returnElement });

            // if this pre-show/hide event was prevented, don't attempt to continue changing the modal state
            if (triggeredEvent.defaultPrevented) {
                return;
            }

            this.returnElement = triggeredEvent.detail.returnElement;
```

This is the change I was talking about.  It sends the current return element to the event listeners, and then applies any changes they make back to the property.  See https://codepen.io/bnickel/pen/oNXoYdX?editors=1011